File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit = 0;
        int ma = prices[prices.size()-1];
        for(int i = prices.size()-1; i>=0; i-=1){
            if (prices[i] >= ma){
                ma = prices[i];
            }

            profit = max(ma - prices[i], profit);
        }

        return profit;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.go
func maxProfit(prices []int) int {
    max_profit := 0
    sell := prices[len(prices)-1]
    for i := len(prices)-2; i >= 0; i-- {
        if prices[i] > sell {
            sell = prices[i]
        }
        profit := sell - prices[i]

        max_profit = max(max_profit, profit)
    }

    return max_profit
}